test/e2e/scheduling/preemption.go

```diff
@@ -394,15 +394,21 @@ var _ = SIGDescribe("SchedulerPreemption [Serial]", func() {
 			// satisfy the pod topology spread constraints. Hence it needs to preempt another low pod
 			// to make the Pods spread like [<high>, <medium, low>].
 			runPausePod(f, mediumPodCfg)
-			e2epod.WaitForPodNotPending(cs, ns, mediumPodCfg.Name)
 
 			ginkgo.By("Verify there are 3 Pods left in this namespace")
 			wantPods := sets.NewString("high", "medium", "low")
 
-			podList, err := cs.CoreV1().Pods(ns).List(context.TODO(), metav1.ListOptions{})
-			pods := podList.Items
+			var pods []v1.Pod
+			err := wait.PollImmediate(framework.Poll, framework.PollShortTimeout, func() (bool, error) {
```

Comment:
Add a comment to explain why we poll instead of doing a one-time check. Also, isn't there any function like this in the framework?

Reply:
Seems there isn't any in the framework :(

Reply:
if we don't need another OWNER approval, I would suggest to just add `WaitForNumberOfRunningPods` or something like that. It would make the test more readable.

Reply:
maybe in a followup, I want it to be merged ASAP so as not to impact the release timeline.

Reply:
sounds good. But just add the comment here. Maybe:

```Wait until the number of pods stabilizes. Note that `medium` pod can get scheduled once the second low priority pod is marked as terminating.```

Reply:
Thanks. Done.